### DISCUSSION TITLE
Configuration merge

### DISCUSSION THREAD

**Discussion by zhaogongchengsi** [web-infra-dev/rspack#4093] [Ideas]
If I need multiple configurations to merge when I use @ rspack/core to call through api, can I provide a mergeConfig api function to merge configuration files?

```ts
const newConfig = mergeConfig(config1, config2, ....)
```

**Comment by hardfist** [comment]
actually webpack supports extends which is for this https://webpack.js.org/configuration/extending-configurations/#extends , we can support this in future

**Reply by hardfist** [reply]
or you can use webpack-merge directly

**Reply by zhaogongchengsi** [reply]
I'm afraid it's not perfectly compatible.

**Reply by hardfist** [reply]
why is it not compatible?

**Reply by dgoldstein0** [reply]
also wondering what's wrong with webpack-merge.  https://rspack.rs/config/#merge-configurations even recommends it so if it's not compatible in some ways the docs should be clarified

**Reply by chenjiahan** [reply]
@dgoldstein0 
webpack-merge is compatible with Rspack config, if you encounter any problems, you can share them with us

**Comment by dgoldstein0** [comment]
related: #11408
